apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js

```diff
@@ -260,231 +108,173 @@ describe('OwnedSectionsTable Sorting', () => {
-      const rowData = sectionRowData[0];
-      const courseLinkCol = shallow(courseLinkFormatter(null, {rowData}));
-      const courseLink = courseLinkCol.find('a').at(0).props().href;
-      const sectionLink = courseLinkCol.find('a').at(1).props().href;
-      assert.equal(
-        courseLink,
-        '//localhost-studio.code.org:3000/courses/csd?section_id=1'
-      );
-      assert.equal(
-        sectionLink,
-        '//localhost-studio.code.org:3000/s/csd1-2019?section_id=1'
+  it('loginInfoFormatter shows the section code for sections managed on Code.org', () => {
+    renderOwnedSectionsTable();
+
+    let hasRowWithThirdParty = false;
+    let hasRowWithSectionCode = false;
+
+    sectionRowData.forEach(section => {
+      let rowSection = null;
+      const loginType = section.loginType;
+
+      if (loginType === SectionLoginType.google_classroom) {
+        // If third party login type, display the provider name rather than the section code
+        rowSection = screen
+          .getByText(i18n.loginTypeGoogleClassroom())
+          .closest('a');
+        expect(screen.queryByText(section.code)).to.be.null;
+        hasRowWithThirdParty = true;
+      } else {
+        // Otherwise, show the section code
+        rowSection = screen.getByText(section.code).closest('a');
+        hasRowWithSectionCode = true;
+      }
+
+      // Check that the button links to the login info tab of the given section
+      assert(
+        rowSection.href.includes(
+          `/teacher_dashboard/sections/${section.id}/login_info`
+        )
       );
     });
 
-    it('courseLinkFormatter contains course text and section text', () => {
-      const rowData = sectionRowData[0];
-      const courseLinkCol = shallow(courseLinkFormatter(null, {rowData}));
-      const courseText = courseLinkCol.find('a').at(0).text();
-      const sectionText = courseLinkCol.find('a').at(1).text();
-      assert.equal(courseText, 'CS Discoveries');
-      assert.equal(sectionText, 'Unit 1: Problem Solving');
-    });
+    // Ensure at least one section has a third party login type and one does not to test
+    // both cases.
+    assert(hasRowWithThirdParty);
+    assert(hasRowWithSectionCode);
+  });
 
-    it('courseLinkFormatter contains button with correct link and text when no course provided', () => {
-      const rowData = sectionRowData[3];
-      const courseLinkCol = shallow(courseLinkFormatter(null, {rowData}));
-      const button = courseLinkCol.text();
-      const link = courseLinkCol.find(Button).prop('href');
-      const text = courseLinkCol.find(Button).prop('text');
-      assert.equal(button, '<Button />');
-      assert.equal(link, '/catalog');
-      assert.equal(text, 'Find a course');
-      expect(
-        courseLinkCol.find({
-          className: skeletonizeContent.skeletonizeContent,
-        })
-      ).to.have.lengthOf(0);
+  it('courseLinkFormatter provides links to course information and section information', () => {
```

Comment:
This test covers a lot! Non-blocking, but it feels like the format of looping through all sections and then checking that 'at least one of them' covers each of a few different states is slightly confusing and not quite future proof. I understand why you combined the concept of checking for assignmentPath counts together, but then it makes it hard to tell which section should have which count. Would it be possible to break this up?

Reply:
Yes for sure! I'll simplify the loop tests so they're more readable and updatable in the future